### DISCUSSION TITLE
why is the logging level set to info for root logger?

### DISCUSSION THREAD

**Discussion by modernNeo** [coddingtonbear/django-mailbox#263] [Q&A] [answered]
I am wondering why, in the `getmail` command, you have set the root logger's config level to INFO instead of setting the config level for `getmail` to INFO like this `logger.setLevel(level=logging.INFO)`

https://github.com/coddingtonbear/django-mailbox/blob/40263b66703332d82c179d79f5ea0d80fc1ea388/django_mailbox/management/commands/getmail.py#L7-L10

**Comment by coddingtonbear** [comment, marked as answer]
That's done so logging messages will be emitted to the console; since that's a command-line entrypoint, some amount of user feedback is desirable. The call to`basicConfig` is doing more than just setting the logging level for the root logger -- it's also wiring things up so they'll be printed to the console.  Using `logging.info` here will allow _all_ `INFO`-level and higher logging messages to make it to the console, regardless of what module they're emitted from.

**Reply by modernNeo** [reply]
so what would you recommend for a user who wants to take advantage of the logic in that `getmail` command and therefore call it from their own code but whose logging setup clashes with the ` logging.basicConfig(level=logging.INFO) `?

**Reply by coddingtonbear** [reply]
It's probably not a great idea to call command-line entrypoints directly from your own logic, but I have to say that just now having glanced at that module, I'm a little surprised about the particular approach I took for setting up the logger.

What's probably the right choice is moving the logic called in the `handler` method into its own function somewhere.  That way, both the command-line entrypoint and user logic (yours) can invoke that logic without you needing to import the command-line class directly.

**Reply by modernNeo** [reply]
> What's probably the right choice is moving the logic called in the handler method into its own function somewhere. That way, both the command-line entrypoint and user logic (yours) can invoke that logic without you needing to import the command-line class directly.

That I agree with. When do you think you can get to that?

**Reply by coddingtonbear** [reply]
Oh, "never" is unfortunately the answer to that question.  If you were to post a PR doing that, though, you'd be one step closer to getting changes like that made.

**Reply by modernNeo** [reply]
Sure, I got no problems making the PR myself. just have to ask you where you would like me to place that bit of logic? a new file or an existing python file in your module?

**Reply by modernNeo** [reply]
@coddingtonbear PR opened: https://github.com/coddingtonbear/django-mailbox/pull/264

**Comment by pfouque** [comment]
Looks solved

**Reply by modernNeo** [reply]
github discussions have to be marked as solved **and** closed?
